File: frontend/src/types/orderStatus.ts
/**
 * Order Status Type Definitions
 *
 * Defines types for custom order status management.
 * Each tenant can define their own order statuses.
 */

/**
 * Order status definition entity
 * Represents a custom order status for a tenant
 */
export interface OrderStatusDefinition {
  id: string;
  tenantId: string;
  name: string; // Display name (e.g., "Pending Payment")
  code: string; // URL-safe code (e.g., "pending-payment")
  color: string; // Hex color code (e.g., "#3B82F6")
  sortOrder: number; // Display order
  isSystemDefault: boolean; // Cannot be deleted if true
  isActive: boolean; // Whether status is available for use
  createdAt: string; // ISO 8601
  updatedAt?: string; // ISO 8601
}

/**
 * Request to create a new order status
 */
export interface CreateOrderStatusRequest {
  name: string;
  code: string;
  color?: string;
  sortOrder: number;
}

/**
 * Request to update an existing order status
 */
export interface UpdateOrderStatusRequest {
  name?: string;
  color?: string;
  sortOrder?: number;
  isActive?: boolean;
}

/**
 * Default status presets for reference
 */
export const DEFAULT_ORDER_STATUSES: Pick<OrderStatusDefinition, 'name' | 'code' | 'color' | 'sortOrder'>[] = [
  { name: 'New', code: 'new', color: '#6B7280', sortOrder: 1 },
  { name: 'Submitted', code: 'submitted', color: '#3B82F6', sortOrder: 2 },
  { name: 'Paid', code: 'paid', color: '#10B981', sortOrder: 3 },
  { name: 'Processing', code: 'processing', color: '#F59E0B', sortOrder: 4 },
  { name: 'Completed', code: 'completed', color: '#059669', sortOrder: 5 },
  { name: 'Cancelled', code: 'cancelled', color: '#EF4444', sortOrder: 6 },
  { name: 'On Hold', code: 'on-hold', color: '#F59E0B', sortOrder: 7 },
  { name: 'Refunded', code: 'refunded', color: '#8B5CF6', sortOrder: 8 },
];

File: api/EComm.Api/Models/OrderStatus.cs
namespace EComm.Api.Models;

/// <summary>
/// Represents a custom order status for a tenant.
/// Each tenant can define their own order statuses with custom names, colors, and ordering.
/// </summary>
public class OrderStatus
{
    public string Id { get; set; } = string.Empty;
    public string TenantId { get; set; } = string.Empty;

    /// <summary>
    /// Display name for the status (e.g., "Pending Payment", "Ready to Ship")
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// URL-safe code for the status (e.g., "pending-payment", "ready-to-ship")
    /// Used in API calls and database queries
    /// </summary>
    public string Code { get; set; } = string.Empty;

    /// <summary>
    /// Hex color code for UI display (e.g., "#3B82F6")
    /// </summary>
    public string Color { get; set; } = "#6B7280"; // Default gray

    /// <summary>
    /// Display order for sorting (lower numbers appear first)
    /// </summary>
    public int SortOrder { get; set; }

    /// <summary>
    /// System default statuses cannot be deleted
    /// </summary>
    public bool IsSystemDefault { get; set; }

    /// <summary>
    /// Whether this status is active and available for use
    /// </summary>
    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

/// <summary>
/// Request to create a new order status
/// </summary>
public class CreateOrderStatusRequest
{
    public string Name { get; set; } = string.Empty;
    public string Code { get; set; } = string.Empty;
    public string Color { get; set; } = "#6B7280";
    public int SortOrder { get; set; }
}

/// <summary>
/// Request to update an existing order status definition
/// </summary>
public class UpdateOrderStatusDefinitionRequest
{
    public string? Name { get; set; }
    public string? Color { get; set; }
    public int? SortOrder { get; set; }
    public bool? IsActive { get; set; }
}

/// <summary>
/// Default order status presets for new tenants
/// </summary>
public static class DefaultOrderStatuses
{
    public static List<(string Name, string Code, string Color, int SortOrder)> GetDefaults() => new()
    {
        ("New", "new", "#6B7280", 1),                    // Gray
        ("Submitted", "submitted", "#3B82F6", 2),        // Blue
        ("Paid", "paid", "#10B981", 3),                  // Green
        ("Processing", "processing", "#F59E0B", 4),      // Orange
        ("Completed", "completed", "#059669", 5),        // Dark Green
        ("Cancelled", "cancelled", "#EF4444", 6),        // Red
        ("On Hold", "on-hold", "#F59E0B", 7),           // Yellow/Orange
        ("Refunded", "refunded", "#8B5CF6", 8)          // Purple
    };
}
